util/etcd/election.go

```diff
@@ -67,7 +67,7 @@ func (eme *MasterElection) Close(ctx context.Context) error {
 	if err := eme.session.Close(); err != nil {
 		glog.Errorf("error closing session: %v", err)
 	}
-	return eme.client.Close()
+	return nil
 }
 
 // ElectionFactory creates etcd.MasterElection instances.
```

Comment:
Maybe add a mention in the comment for `NewElectionFactory` that the passed in `client` should remain valid for the lifetime of the `ElectionFactory` ?